src/parser/WASMParser.cpp

```diff
@@ -567,24 +571,27 @@ class WASMBinaryReader : public wabt::WASMBinaryReaderDelegate {
         return info;
     }
 
-    size_t popVMStack()
+    VMStackInfo& peekVMStackInfo()
     {
-        return popVMStackInfo().position();
+        // FIXME This error can occur during the parsing process because of invalid wasm instructions
+        // e.g. a function with no end opcode
```

Comment:
If the code is valid, this should be dead code, isn''t it?

Reply:
If the wasm code is valid, this assert would be useless as you mentioned.
But this assert is newly added for invalid wasm code.
Since we generate bytecodes one by one during the parsing of wasm code, invalid wasm instructions might trigger incorrect behavior like pop of empty vmStack. So, I added this assert to simply check the invalid case as we basically assume that Walrus handles only valid wasm codes.